resources/vault_installation_binary.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Cookbook: hashicorp-vault
+# License: Apache 2.0
+#
+# Copyright 2015-2016, Bloomberg Finance L.P.
+#
+
+resource_name :vault_installation_binary
+
+property :version, String
+property :archive_url, String
+property :archive_basename, String, name_property: true
+property :archive_checksum, String
+property :extract_to, String, default: '/opt/vault'
+
+default_action :create
+
+action :create do
+  [new_resource.extract_to, ::File.join(new_resource.extract_to, new_resource.version)].each do |path|
+    directory path do
+      mode '0755'
+      recursive true
+    end
+  end
+
+  # Still coming from poise
```

Comment:
Not sure if we should use ark here? Trying to replace poise wherever possible. Also this is completely untested compared to the previous zip functionality.